platform/broadcom/sonic-platform-modules-delta/ag9032v2a/modules/delta_ag9032v2a_platform.c

```diff
@@ -636,21 +636,21 @@ static ssize_t set_lpmode_data(struct device *dev, struct device_attribute *dev_
         return err;
     }
     mutex_lock(&dni_lock);
-    /*QSFP1~8*/
-    set_bytes = reverse_8bits(set_data & 0xff);
-    i2c_smbus_write_byte_data(pdata[swpld1].client, QSFP_LPMODE_1, set_bytes);
-
-    /*QSFP9~16*/
-    set_bytes = reverse_8bits((set_data >> 8 ) & 0xff);
-    i2c_smbus_write_byte_data(pdata[swpld1].client, QSFP_LPMODE_2, set_bytes);
+    /*QSFP25~32*/
+    set_bytes = set_data & 0xff;
+    i2c_smbus_write_byte_data(pdata[swpld1].client, QSFP_LPMODE_4, set_bytes);
 
     /*QSFP17~24*/
-    set_bytes = reverse_8bits((set_data >> 16 ) & 0xff);
+    set_bytes = (set_data >> 8 ) & 0xff;
     i2c_smbus_write_byte_data(pdata[swpld1].client, QSFP_LPMODE_3, set_bytes);
 
-    /*QSFP25~32*/
-    set_bytes = reverse_8bits((set_data >> 24 ) & 0xff);
-    i2c_smbus_write_byte_data(pdata[swpld1].client, QSFP_LPMODE_4, set_bytes);
+    /*QSFP9~16*/
+    set_bytes = (set_data >> 16 ) & 0xff;
+    i2c_smbus_write_byte_data(pdata[swpld1].client, QSFP_LPMODE_2, set_bytes);
+
+    /*QSFP1~8*/
+    set_bytes = (set_data >> 24 ) & 0xff;
+    i2c_smbus_write_byte_data(pdata[swpld1].client, QSFP_LPMODE_1, set_bytes);
     mutex_unlock(&dni_lock);
     return count;
```

Comment:
where are we getting the "count" value? the return type is ssize_t but count is size_t.

Reply:
This platform is x86, ssize_t is the same as size_t, but is a signed type - read ssize_t as “signed size_t”. ssize_t is able to represent the number -1,  which is returned as a way to indicate error.
this usage follow https://github.com/Azure/sonic-buildimage/blob/master/platform/broadcom/sonic-platform-modules-accton/as5812-54t/modules/x86-64-accton-as5812-54t-cpld.c +216

Reply:
But `count` is passed into the function, and it is not modified in the function. How does it get set to -1 to signify an error?

Reply:
@JohnsonYJLu if i2c_smbus_write_byte_data function get error result, then it goes to ERROR statement and it will return the status value. So for write success case, this will just return the passing count value as the returning value.